test/parallel/test-buffer-safe-unsafe.js

```diff
@@ -7,8 +7,18 @@ const safe = Buffer.alloc(10);
 
 function isZeroFilled(buf) {
   for (let n = 0; n < buf.length; n++)
-    if (buf[n] > 0) return false;
+    if (buf[n] !== 0) return false;
```

Comment:
Looking at it now this one should have been obvious :-/. The bug only shows up after doing a pooled unsafeAlloc allocation because the zero fill flag was never being reset (because a real allocation wasn't being done).